sys/net/gnrc/network_layer/ipv6/ext/gnrc_ipv6_ext.c

```diff
@@ -20,12 +20,17 @@
 
 #include "net/gnrc/ipv6/ext.h"
 
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
 bool gnrc_ipv6_ext_demux(kernel_pid_t iface, gnrc_pktsnip_t *pkt,
                          uint8_t nh)
 {
-    gnrc_pktsnip_t *ext_snip;
+    gnrc_pktsnip_t *ext_snip, *tmp;
     ipv6_ext_t *ext;
     unsigned int offset = 0;
+    ipv6_hdr_t *hdr;
+    int res;
 
     ext = ((ipv6_ext_t *)(((uint8_t *)pkt->data) + sizeof(ipv6_hdr_t)));
```

Comment:
If `pkt` is referred from multiple pointers, `gnrc_pktbuf_start_write` duplicates the packet data:

![pkt](https://cloud.githubusercontent.com/assets/3772478/13104011/e9e25466-d59d-11e5-8a4f-77011c9d54af.png)

`ext` must point to the new byte array.

Note that the buffer referred from `ext` is updated by this line:
https://github.com/RIOT-OS/RIOT/pull/4774/files#diff-6906777c73d641182b398573d3bdba0bR87